python/Makefile.am

```diff
@@ -14,9 +14,37 @@ EXTRA_DIST = pylintrc
 SUBDIRS	= pacemaker \
 	  tests
 
+# There are two things going on with this check-local target:
+#
+# (1) The spaces before "ifeq" and "endif" are intentional and required.  Do not
+#     remove them.  Without those spaces, automake/autoconf will attempt to parse
+#     the conditionals and error out with "error: else without if".  A space is
+#     enough to fool it.  I expect this will break at some point in the future.
+#
+# (2) The reason for the conditional in the first place is that we build the
+#     pacemaker python module from two pieces:  one part is auto-generated from .in
+#     files, and one part is not.  If we are doing a build with a separate build
+#     directory, the auto-generated bits will end up in the build directory, but the
+#     other bits will remain in the source directory.
+#
+#     Python sees this as a pacemaker module in the build directory and a pacemaker
+#     module in the source directory.  Whichever one is in the path first wins, and
+#     python won't be able to find the files in the other directory.
+#
+#     So for instance, let's say we have python/pacemaker/exitstatus.py and
+#     _build/python/pacemaker/buildoptions.py (generated from buildoptions.py.in).
+#     If we set PYTHONPATH=python:_build/python, it will be able to import exitstatus.py
+#     but not buildoptions.py.
+#
+#     It's not clear there's anything that can be done to fix this situation, aside
+#     from copying everything from one spot to the other.  On the other hand, we
+#     could just disable running tests if there's a separate build directory.  This
+#     isn't ideal, but they'll probably still be run enough in development to catch
+#     any problems.
 .PHONY: check-local
-check-local:
-	PYTHONPATH=$(top_srcdir)/python:$(top_builddir)/python $(PYTHON) -m unittest discover -v -s $(top_srcdir)/python/tests $(top_builddir)/python/tests
+ ifeq ($(top_srcdir),$(top_builddir))
```

Comment:
This is not working as you think it does. None of the logs in CI show the call to unittest.

Instead of risking with ifeq that can potentially break and might be dependent on the automake version of the day, you can use a much safer approach using shell.

if [ "$(top_srcdir)" = "$(top_builddir)" ]; then \
 PYTHONPATH=$(top_srcdir)/python $(PYTHON) -m unittest discover -v -s $(top_srcdir)/python/tests \
else \
 echo "out of tree build detected, PYTHON test not executed" \
fi;

_or_ something like that.  they might be missing some ; here and there.

Reply:
Oh, I accidentally deleted one line too many.  This version appears to work as it should in mock, when builddir == srcdir, and when builddir != srcdir.

I'm not actually concerned about automake - I got the idea from a kernel patch in 2015, and it's still working.   I don't really like using shell in makefiles unless I absolutely have to.